src/main/java/net/minecraftforge/client/model/animation/Animation.java

```diff
@@ -0,0 +1,186 @@
+package net.minecraftforge.client.model.animation;
+
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import net.minecraftforge.client.model.IModelState;
+import net.minecraftforge.client.model.animation.ModelBlockAnimation.MBJointInfo;
+import net.minecraftforge.common.property.IUnlistedProperty;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableTable;
+
+public class Animation
+{
+    public static final IUnlistedProperty<IModelState> AnimationProperty = new IUnlistedProperty<IModelState>()
+    {
+        public String getName() { return "forge_animation"; }
+        public boolean isValid(IModelState state) { return true; }
+        public Class<IModelState> getType() { return IModelState.class; }
+        public String valueToString(IModelState state) { return state.toString(); }
+    };
+
+    public static float getWorldTime(World world)
+    {
+        return getWorldTime(world, 0);
+    }
+
+    public static float getWorldTime(World world, float tickProgress)
+    {
+        return (world.getTotalWorldTime() + tickProgress) / 20;
+    }
+
+    private static final AnimationStateMachine missing = new AnimationStateMachine(
```

Comment:
that treats the missing as a special case, as opposed this which simply defines missing as a normal case that is empty. We generally want to avoid creating special cases in code

If you used Optional, you would have to wrap and unwrap every. single. place. you wanted to use